plugins/fileservices/fileservices.cpp

```diff
@@ -2747,21 +2751,60 @@ static void checkExternalFileRights(ICodeContext *ctx, CDfsLogicalFileName &lfn,
     }
 }
 
-FILESERVICES_API void  FILESERVICES_CALL fsMoveExternalFile(ICodeContext * ctx,const char *location,const char *frompath,const char *topath)
+static void checkExternalFilePath(ICodeContext *ctx,IPropertyTree *plane,const char *host,
+    const char *path,bool checkScopePermissionIfDir,bool rd,bool wr,RemoteFilename &rfn)
 {
-    SocketEndpoint ep(location);
+    if (containsRelPaths(path)) //Detect a path like: a/../../../f
+        throw makeStringExceptionV(-1,"Invalid file path %s",path);
+
+    CDfsLogicalFileName dlfn;
+    if (plane)
+    {
+        //allow relative path if a plane name is specified.
+        if (isAbsolutePath(path))
+        {
+            const char *relativePath = getRelativePath(path,plane->queryProp("@prefix"));
+            if (nullptr == relativePath)
+                throw makeStringExceptionV(-1,"Invalid plane path %s.",path);
+            path = relativePath;
+        }
+        dlfn.setPlaneExternal(plane->queryProp("@name"),path);
+    }
+    else
+        dlfn.setExternal(host,path); //for backward compatibility
+    dlfn.getExternalFilename(rfn);
+
+    if (checkScopePermissionIfDir)
```

Comment:
see prev. comment. I think this should be done in implementMoveExternalFile, ensuring the lfn is always to the dir.